Design/ThirdpartyDependencies.md

```diff
@@ -0,0 +1,300 @@
+are provided for developers to easily install the required dependencies.
+
+Upgrading libraries is possible by placing newer versions in the above locations.
+For some libraries, such as HDF5 or sip, 
+this is not advised due to conflicts with other applications requiring
+different versions.
+A real example is with the rebuilt `sip` package for RHEL7,
+which conflicts with other packages such as `Octave` and causes users troubles
+when installing these packages alongside Mantid.
+
+The package provided for users depends on the system libraries leaving the
+mantid package to just ship the mantid libraries.
+
+In addition, 
+for RHEL 7 (and its variants CentOS, Scientfic Linux) a
+[Software Collection](https://www.softwarecollections.org/en/) is used to provide
+a gcc-7 allowing C++17-enabled builds.
+
+Pros:
+* Easier developer setup.
+* Smaller user packages. The system libraries are shared for all packages resulting
+  in a much smaller package to install.
+* (Mostly) managed for us.
+* Simply commands to install packages (apt/yum)
+
+
+Cons:
+* Dependencies not versioned with the main code. Hard to track versions used.
+* Versions become outdated very quickly. Facilities tend to use
+  older distributions that do not have up to date versions of packages.
+* Upgrading/rebuilding package versions can cause conflicts with other packages
+  not related to mantid and even old mantid versions.
+* New dependencies have to be installed manually on developers and builders machines.
+
+
+### Windows
+
+Windows uses a collection of custom-built libraries managed
+by a homegrown set of [scripts](https://github.com/mantidproject/thirdparty-msvc2015).
+The binaries are stored using the [git lfs](https://git-lfs.github.com/)
+extension for managing large binaries with git.
+For developers, all of the libraries are pulled down by
+the [cmake configure step](https://github.com/mantidproject/mantid/blob/master/buildconfig/CMake/Bootstrap.cmake#L13)
+thereby minimizing the effort required for developers to get up and running.
+It also includes a bundled version of Python as none is available by default
+on Windows.
+
+The package provided to users contains all required dependencies and is a
+self-contained bundle.
+
+Pros:
+* Easy developer setup.
+* Complete freedom to choose package versions and apply patches etc.
+* New dependencies are retrieved automatically for developers and build servers
+* Self-contained user package.
+
+Cons:
+* Dependencies not versioned with the main code. Hard to track versions used.
+* Homegrown set of scripts are very clunky.
+* Maintenance of build scripts adds overhead.
+* Complete bundling of dependencies gives a much larger user installer package.
+
+
+### MacOS
+
+MacOS uses the [Homebrew](https://brew.sh/) package manager along with a custom
+set of [formulae](https://github.com/mantidproject/homebrew-mantid) for obselete
+dependencies.
```

Comment:
I would also mention explicitly that `pip` is used to manage all the python dependencies.